### PR TITLE
FT: Add doc for replication to AWS setup

### PR DESCRIPTION
**Pull request by bennettbuchanan**
Add walkthrough documentation for setting up replication to AWS.

### CODE REVIEW

File: docs/crr-to-aws-s3.md [hunk 1]
```diff
@@ -75,3 +75,334 @@ env_replication_endpoints:
...
+    --bucket <target-bucket> \
+    --versioning-configuration Status=Enabled \
+    --profile <target-profile>
+    ```
+
+3. Generate account credentials for the source site using the command:
+
+    ```
+    ansible-playbook -i env/client-template/inventory \
+    tooling-playbooks/generate-account-accesskey.yml
```
Comment (jonathan-gramain) [inline docs/crr-to-aws-s3.md]: *`generate-account-access-key.yml`

File: docs/crr-to-aws-s3.md [hunk 2]
```diff
@@ -75,3 +75,334 @@ env_replication_endpoints:
...
+    ```
+
+4. At the source, log in to the docker shell of a Backbeat container:
+
+    ```
+    docker exec -it scality-backbeat-1 bash
```
Comment (jonathan-gramain) [inline docs/crr-to-aws-s3.md]: On light-local it appears as `scality-backbeat-127.0.0.2-1`, not sure on a full stack deployment but it could be the same.

File: docs/crr-to-aws-s3.md [hunk 3]
```diff
@@ -75,3 +75,334 @@ env_replication_endpoints:
...
+    ```
+
+    ```
+    aws s3api get-bucket-versioning \
+    --bucket <target-bucket> \
+    --profile <target-profile>
+    ```
+
+3. Establish a bucket policy to allow the Backbeat replication engine to
+    retrieve the appropriate information. Create the file _policy.json_ with the
```
Comment (jonathan-gramain) [inline docs/crr-to-aws-s3.md]: Not only retrieve, but also write data/metadata. Maybe rephrase to `to execute the required actions`?

File: docs/crr-to-aws-s3.md [hunk 4]
```diff
@@ -75,3 +75,334 @@ env_replication_endpoints:
...
+                ]
+            },
+            {
+                "Effect": "Allow",
+                "Action": [
+                    "s3:ReplicateObject",
+                    "s3:ReplicateDelete",
+                    "s3:ReplicateTags"
+                ],
+                "Resource": "arn:aws:s3:::bennett-federationtest/*"
```
Comment (jonathan-gramain) [inline docs/crr-to-aws-s3.md]: For an example AWS bucket name, maybe use a more self-explanatory name like `bennett-aws-bucket`? Or stick with the generic `<target-bucket>`. Maybe same applies above for the source bucket.

File: docs/crr-to-aws-s3.md [hunk 5]
```diff
@@ -75,3 +75,334 @@ env_replication_endpoints:
 
 * Backbeat updates the source object's metadata to store AWS' version id and set
   replication status as `COMPLETED`
+
+## Setting up Cross-Region Replication (CRR) to AWS
+
+This section contains steps to set up cross-region replication from Scality S3
+to Amazon Web Services S3 using the Backbeat replication engine. Two subsections
+are provided with different methods: the subsection "Basic Example" uses a
```
Comment (jonathan-gramain) [inline docs/crr-to-aws-s3.md]: The explanation about the subsections may go directly into their respective "Basic" and "Detailed" subsections introductory paragraph IMO. And instead of "Basic" and "Detailed", names like "Half-Automated" and "Manual" (for example) seem more appropriate to me.

**Review by jonathan-gramain — COMMENTED** [review]
Great to see both the script version and manual version, so the user can see what's going on under the hood if he wants too :)

A few minor comments

File: docs/crr-to-aws-s3.md (hunk 1 above)
Comment (bennettbuchanan) [reply docs/crr-to-aws-s3.md]: Nice-catch. Thanks!

File: docs/crr-to-aws-s3.md (hunk 2 above)
Comment (bennettbuchanan) [reply docs/crr-to-aws-s3.md]: On a full stack deployment the name of the docker container is `scality-backbeat-1`. I will add a note here about the difference.

File: docs/crr-to-aws-s3.md (hunk 4 above)
Comment (bennettbuchanan) [reply docs/crr-to-aws-s3.md]: Yes, this is meant to be `<target-bucket>`! That is left over from when I was testing :smile:.

File: docs/crr-to-aws-s3.md (hunk 5 above)
Comment (bennettbuchanan) [reply docs/crr-to-aws-s3.md]: I like that idea! Updated.

File: docs/crr-to-aws-s3.md (hunk 3 above)
Comment (bennettbuchanan) [reply docs/crr-to-aws-s3.md]: Agree—that is better wording. Updated.

**Review by philipyoo — APPROVED** [review]

File: docs/crr-to-aws-s3.md [hunk 6]
```diff
@@ -75,3 +75,340 @@ env_replication_endpoints:
...
+    --profile <source-profile>
+    ```
+
+    ```
+    aws s3api get-bucket-versioning \
+    --bucket <target-bucket> \
+    --profile <target-profile>
+    ```
+
+3. Establish a bucket policy to allow the Backbeat replication engine to
```
Comment (rahulreddy) [inline docs/crr-to-aws-s3.md]: Let's not call it a bucket policy since bucket policies are a whole different game. `IAM resource policy` would be correct.
```
Establish an IAM resource policy on the source bucket to allow the Backbeat to
execute the required actions.
```

**Review by rahulreddy — CHANGES_REQUESTED** [review]

File: docs/crr-to-aws-s3.md [hunk 7]
```diff
@@ -75,3 +75,340 @@ env_replication_endpoints:
...
+    ```
+    [<source-profile>]
+    aws_access_key_id = <access-key>
+    aws_secret_access_key = <secret-key>
+    ```
+
+6. Set up replication on your buckets:
+
+    ```
+    node ~/replication/backbeat/bin/setupReplication.js setup \
```
Comment (bennettbuchanan) [inline docs/crr-to-aws-s3.md]: The path should be `bin/replication.js`
